hw/ip/entropy_src/doc/_index.md

```diff
@@ -104,70 +166,45 @@ After power-up, the ENTROPY_SRC block is disabled.
-The HWIF FIFO has lower priority than the TLUL FIFO, but is expected to have higher bandwidth demands.
-Because of this, higher priority is given to the TLUL FIFO so that it does not starve when the hardware entropy interface is consuming a high level of entropy bits.
-
-The {{< regref "ES_THRESH" >}} register is used to set the point of notification to the firmware.
-Notification can be either by polling or by an interrupt.
-For debug, the {{< regref "ES_FDEPTHST" >}} register can be read to find out the current state of the TLUL FIFO at any time.
-
-An additional feature is the {{< regref "ES_RATE" >}} register.
-The purpose of this register is to simulate faster or slower entropy generation sources when is digial source mode.
-This will aid firmware design and debug when trying to handle all entropy source types and rates.
-When in rng mode, this register must exactly match the sample rate required by the rng source, expected to be in the range of about 25 to 50 kilohertz.
+The `es_entropy_valid` bit will indicate to the firmware when entropy bits can read from the {{< regref "ENTROPY_DATA" >}} register.
+The firmware will do 32-bit register reads of the {{< regref "ENTROPY_DATA" >}} register to retrieve the entropy bits.
+Each read will automatically pop an entry from the entropy unpacker block.
+A full twelve 32-bit words need to be read at a time.
 
-### Interrupts
+The hardware entropy interface will move entropy bits out of the ENTROPY FIFO when it is not empty, and the downstream hardware is ready.
+If firmware is not currently reading entropy bits, all processed entropy bits will flow to the hardware entropy interface.
 
-The ENTROPY_SRC module has several interrupts: `es_entropy_valid`, `es_rng_src_ok`, es_rct_failed, es_apt_failed`, and `es_fifo_err`.
+An additional feature is the {{< regref "RATE" >}} register.
+The purpose of this register is to simulate faster or slower entropy generation sources than when the digital source is selected.
+This will aid firmware design and debug when trying to handle all entropy source types and rates.
+When in PTRNG mode, this register must exactly match the sample rate required by the PTRNG source, expected to be in the range of about 25 to 50 kilohertz.
 
-The `es_entropy_valid` interrupt should be used when a entropy source has been implemented that is relatively slow.
+### Interrupts
 
-The `es_rng_src_ok` interrupt should be used when rng entropy mode has been selected.
-Once the external entropy source is operational, this interrupt will be triggered.
+The ENTROPY_SRC module has several interrupts: `es_entropy_valid`, es_health_test_failed`, and `es_fifo_err`.
 
-The `es_rct_failed` interrupt will trigger when the internal health test fails the repetitive count test.
+The `es_entropy_valid` interrupt should be asserted when an entropy source has been implemented that is relatively slow.
```

Comment:
Maybe: "The es_entropy_valid interrupt is designed to be used when a relatively slow entropy source is connected.  It will trigger when a sample has been collected."  (I'm not actually sure what the correct phrasing is for the trigger condition here, is it a sample, a word, or something else?)